doc/source/whatsnew/v0.25.0.rst

```diff
@@ -136,7 +136,7 @@ Categorical
 Datetimelike
 ^^^^^^^^^^^^
 
--
+- Bug with :class:`Timestamp` and :class:`CustomBusinessDay` arithmetic throwing an exception with datetime subclasses (:issue:`25734`)
```

Comment:
we don't throw an exception, rather we raise an exception. Can you even be more explicit here, what does this have to do with CBD?

Reply:
I tried to scope this description to more what was in the original issue, but really it is due to `freezegun` monkeypatching `datetime` with a subclass of `datetime`.  A few of the `offsets.py` classes turn things into `datetime` in their `apply` functions.  That's why this affects `CustomBusinessDay` and not, for instance, `BusinessDay`.  After a brief look, a few other classes might be affected by this bug, like `Easter` and `FY5253`.

However, in writing a custom `datetime` class for testing, another bug was discovered in `convert_datetime_to_tsobject` which prevents *any* subclassed `datetime` from being converted into a `Timestamp`.  `freezegun` was not affected by this by chance because it happened to implement the `nanosecond` property in their custom `datetime` class, but that fix has been included in this PR as mentioned in another comment.

Reply:
ok, so say this affects subclasses of datetime. note btw, we don't provide *any* compat with this, but as usual will take patches.

Reply:
The fact that this affects datetime subclasses is already in the release notes, I'm not sure what you would like changed.

Reply:
> :class:`CustomBusinessDay` arithmetic is misleading